runtime/executor/src/main/java/edu/snu/nemo/runtime/executor/TaskGroupExecutor.java

```diff
@@ -257,16 +257,28 @@ private void launchBoundedSourceTask(final BoundedSourceTask boundedSourceTask)
    */
   private void launchOperatorTask(final OperatorTask operatorTask) {
     final Map<Transform, Object> sideInputMap = new HashMap<>();
+    final List<DataUtil.IteratorWithNumBytes> sideInputIterators = new ArrayList<>();
     final String physicalTaskId = getPhysicalTaskId(operatorTask.getId());
+
     final Map<String, Object> metric = new HashMap<>();
     metricCollector.beginMeasurement(physicalTaskId, metric);
+    long accumulatedBlockedReadTime = 0;
+    long accumulatedWriteTime = 0;
+    long accumulatedSerializedBlockSize = 0;
+    long accumulatedEncodedBlockSize = 0;
+    boolean blockSizeAvailable = true;
 
     final long readStartTime = System.currentTimeMillis();
     // Check for side inputs
     physicalTaskIdToInputReaderMap.get(physicalTaskId).stream().filter(InputReader::isSideInputReader)
         .forEach(inputReader -> {
           try {
-            final Object sideInput = inputReader.getSideInput();
+            if (!inputReader.isSideInputReader()) {
+              throw new RuntimeException();
```

Comment:
Please add some description about this exception.

Reply:
Good idea. Done.